pulsar-impl/src/main/java/com/datastax/oss/sink/pulsar/AvroTypeUtil.java

```diff
@@ -15,24 +15,74 @@
  */
 package com.datastax.oss.sink.pulsar;
 
+import java.nio.ByteBuffer;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import org.apache.avro.Conversion;
+import org.apache.avro.Schema;
 import org.apache.pulsar.client.api.schema.GenericRecord;
 import org.apache.pulsar.common.schema.SchemaType;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public final class AvroTypeUtil {
 
-  private static final Logger log = LoggerFactory.getLogger(AvroTypeUtil.class);
+  private static Map<String, Conversion<?>> logicalTypeConverters = new HashMap<>();
+
+  static {
+    logicalTypeConverters.put(
+        CqlLogicalTypes.CQL_DECIMAL, new CqlLogicalTypes.CqlDecimalConversion());
+    logicalTypeConverters.put(
+        CqlLogicalTypes.CQL_DURATION, new CqlLogicalTypes.CqlDurationLogicalType());
+    logicalTypeConverters.put(
+        CqlLogicalTypes.CQL_VARINT, new CqlLogicalTypes.CqlVarintConversion());
+  }
 
   private AvroTypeUtil() {}
 
   public static boolean shouldWrapAvroType(GenericRecord record, Object fieldValue) {
     return record != null && record.getSchemaType() == SchemaType.AVRO && isMapOrList(fieldValue);
   }
 
+  public static boolean shouldHandleLogicalType(GenericRecord record, Object fieldValue) {
+    if (record == null && record.getSchemaType() != SchemaType.AVRO) {
+      return false;
+    }
+    if (isVarint(fieldValue)) {
```

Comment:
we should have access to the schema somehow here
we should use the Schema to safely decode these objects